tuftool/src/root.rs

```diff
@@ -275,16 +283,34 @@ impl Command {
         write_file(path, &root)
     }
 
-    fn sign(path: &PathBuf, key_source: Box<dyn KeySource>) -> Result<()> {
+    fn sign(
+        path: &PathBuf,
+        key_source: &[Box<dyn KeySource>],
+        cross_sign: Option<PathBuf>,
+    ) -> Result<()> {
         let root: Signed<Root> = load_file(path)?;
 
-        let signed_root = SignedRole::new(
-            root.signed.clone(),
-            &KeyHolder::Root(root.signed),
-            &[key_source],
-            &SystemRandom::new(),
-        )
-        .context(error::SignRoot { path })?;
+        let signed_root = match cross_sign {
+            // This will get the keys from current root.json for validation
+            None => SignedRole::new(
+                root.signed.clone(),
+                &KeyHolder::Root(root.signed),
+                key_source,
+                &SystemRandom::new(),
+            )
+            .context(error::SignRoot { path })?,
+            Some(cross_sign_root) => {
+                // This will get the keys from cross-sign root.json for validation
+                let old_root: Signed<Root> = load_file(&cross_sign_root)?;
+                SignedRole::new(
+                    root.signed,
+                    &KeyHolder::Root(old_root.signed),
+                    key_source,
+                    &SystemRandom::new(),
+                )
+                .context(error::SignRoot { path })?
+            }
+        };
```

Comment:
Something like this (pseudo-code) might be a little DRY-er.

```suggestion
        let loaded_root = match cross_sign {
            None => root,
            Some(cross_sign_root) => load_file(&cross_sign_root)?
        };
        let signed_root = SignedRole::new(
            root.signed,
            &KeyHolder::Root(loaded_root),
            key_source,
            &SystemRandom::new(),
        )
        .context(error::SignRoot { path })?
```

Reply:
Yup changed it in new commit.